kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/TemplateKekSelector.java

```diff
@@ -58,15 +58,11 @@ public CompletionStage<Map<String, K>> selectKek(@NonNull Set<String> topicNames
                     .map(
                             topicName -> kms.resolveAlias(evaluateTemplate(topicName))
                                     .exceptionallyCompose(e -> {
-                                        if (e instanceof CompletionException && e.getCause() != null) {
-                                            e = e.getCause();
-                                        }
-                                        if (e instanceof UnknownAliasException) {
+                                        if (e instanceof UnknownAliasException
+                                                || (e instanceof CompletionException ce && ce.getCause() instanceof UnknownAliasException)) {
```

Comment:
In a true break from tradition for me, I wonder if (in some other PR) we should introduce a util method to check if a future exception is an X. 
For example this would become 
```suggestion
if (ExceptionUtils.is(e, UnknownAliasException.class)) {
```

and exception utils looks something like 

```java
public boolean is(Exception e, Class<?> targetType) {
    return targetType.isAssignableFrom(e.getClass()) || (CompletionException.class.isAssignableFrom(e) && targetType.isAssignableFrom(e.getCause().getClass())
}
```
Note this has not met a compiler...

Reply:
yes, that sounds like a good idea.  I expect we'll meet the same pattern over and over.